src/rum/core.cljs

```diff
@@ -106,7 +107,13 @@
           (this-as this
             (vswap! (state this) call-all did-update)))))
 
-    (gobj/set prototype "componentWillUnmount"            
+    (when-not (empty? did-catch)
+      (gobj/set prototype "componentDidCatch"
+        (fn [error info]
+          (this-as this
+            (vswap! (state this) #(call-all %1 %2 error info) did-update)))))
```

Comment:
I believe you've meant `did-catch` here.

Also, you could get rid of anonymous function:

```
(vswap! (state this) call-all did-catch error info)
```